.github/workflows/reusable_provenance.yaml

```diff
@@ -144,21 +150,34 @@ jobs:
           if [[ "${{ github.event_name }}" != "pull_request" ]]; then
             provenance_file="${provenance_file}.build.slsa"
           fi
+          package_name="${{ needs.get_inputs.outputs.package-name }}"
           provenance_digest="$(ent put --digest-format=human --porcelain "${provenance_file}")"
-          for file in ${{ needs.get_inputs.outputs.artifact-path }}; do
-              if [[ -f "${file}" ]]; then
-                  binary_name="$(basename "${file}")"
-                  binary_digest="$(ent put --digest-format=human --porcelain "${file}")"
-                  ent tag set --public-key=${{ inputs.ent-public-key }} \
-                    --label="artifact_${GITHUB_SHA}_${binary_name}" \
-                    --target="${binary_digest}"
-                  ent tag set --public-key=${{ inputs.ent-public-key }} \
-                    --label="provenance_${GITHUB_SHA}_${binary_name}" \
-                    --target="${provenance_digest}"
-              else
-                  echo "Warning: Input '${file}' is not a file."
-              fi
-          done
+          artifact_count=$(ls -1 ${{ needs.get_inputs.outputs.artifact-path }} | wc -l)
+          if [ "${artifact_count}" -eq 1 ]; then
+            binary_file=${{ needs.get_inputs.outputs.artifact-path }}
+            binary_digest="$(ent put --digest-format=human --porcelain "${binary_file}")"
+            ent tag set --public-key=${{ inputs.ent-public-key }} \
+              --label="artifact_${GITHUB_SHA}_${package_name}" \
+              --target="${binary_digest}"
+            ent tag set --public-key=${{ inputs.ent-public-key }} \
+              --label="provenance_${GITHUB_SHA}_${package_name}" \
+              --target=${provenance_digest}
+          else
```

Comment:
this would leak to unexpected behavior in cases where the artifact-path is a wildcard, but it happens to resolve to only one file.